drivers/SmartThings/philips-hue/src/logjam.lua

```diff
@@ -1,9 +1,84 @@
 local log = require "log"
+local st_utils = require "st.utils"
+-- trick to fix the VS Code Lua Language Server typechecking
+---@type fun(val: any?, name: string?, multi_line: boolean?): string
+st_utils.stringify_table = st_utils.stringify_table
 
 local logjam = {}
+
+logjam.real_log = log.log
+
+logjam.enabled_modules = {}
+
+function logjam.inject_global()
+  for field_key, level_key in pairs(log) do
+    if
+        string.find(field_key, "LOG_LEVEL_") and
+        type(level_key) == "string" and
+        type(log[level_key]) == "function"
+    then
+      log[level_key] = logjam[level_key]
+      local level_with_key = string.format("%s_with", level_key)
+      if type(log[level_with_key]) == "function" then
+        log[level_with_key] = logjam[level_with_key]
+      end
+    end
+  end
+  log.log = logjam.log
+end
+
+function logjam.enable_passthrough()
+  logjam.passthrough = true
+end
+
+function logjam.disable_passthrough()
+  logjam.passthrough = false
+end
+
+function logjam.enable(module)
+  logjam.enabled_modules[module] = true
+end
+
+function logjam.disable(module)
+  logjam.enabled_modules[module] = nil
+end
+
 function logjam.log(opts, level, ...)
-  if opts.on == true then
-    log.log(opts, level, ...)
+  local call_info
+  if not opts.call_info then
+    call_info = debug.getinfo(2)
+  else
+    call_info = opts.call_info
+  end
+  opts.call_info = nil
+
+  local module_name = nil
+  if type(call_info.source) == "string" then
+    module_name =
+    call_info.source
+    :gsub("%.lua", "")
+    :gsub("/init", "")
+    :gsub("/", ".")
+    :gsub("^init$", "philips-hue")
```

Comment:
nit: indentation here feels a bit off
```suggestion
    module_name = call_info.source
      :gsub("%.lua", "")
      :gsub("/init", "")
      :gsub("/", ".")
      :gsub("^init$", "philips-hue")
```